app/src/main/java/com/humberto/tasky/agenda/presentation/agenda_details/AgendaDetailsAction.kt

```diff
@@ -14,4 +14,5 @@ sealed interface AgendaDetailsAction {
     data class OnSelectFromTime(val fromTime: LocalTime): AgendaDetailsAction
     data class OnSelectToTime(val toTime: LocalTime): AgendaDetailsAction
     data class OnSelectAtTime(val atTime: LocalTime): AgendaDetailsAction
+    data class OnSelectReminderAt(val remindAt: Long): AgendaDetailsAction
```

Comment:
I found it better to use an enum through my presentation and domain layer to represent when the user wants an alarm to go off for an agenda item. Then, I calculate the Long value to send to the API in my data layer.

With this approach, I don't perform any calculations (to get a new Long value) when the user changes the reminder time. I also get the advantage of working with a higher-level object like an enum instead of a Long.

```
import kotlin.time.Duration
import kotlin.time.Duration.Companion.days
import kotlin.time.Duration.Companion.hours
import kotlin.time.Duration.Companion.minutes

enum class RemindBeforeDuration(val duration: Duration) {
  TEN_MINUTES(10.minutes),
  THIRTY_MINUTES(30.minutes),
  ONE_HOUR(1.hours),
  SIX_HOURS(6.hours),
  ONE_DAY(1.days)
}
```